packages/token-sdk/src/token-fetcher.ts

```diff
@@ -9,23 +9,33 @@ import {
 } from "@orca-so/common-sdk";
 import { MintInfo } from "@solana/spl-token";
 import invariant from "tiny-invariant";
-import { MetadataProvider, MetadataUtil } from "./metadata";
+import { MetadataProvider, MetadataUtil, TokenMetadata } from "./metadata";
+import pTimeout from "p-timeout";
+
+const TIMEOUT_MS = 2 * 60 * 1000; // 2 minutes
+
+interface Opts {
+  timeoutMs?: number;
+  cache?: Record<string, Token>;
+}
 
 type ReadonlyToken = Readonly<Token>;
 type ReadonlyTokenMap = Readonly<Record<string, ReadonlyToken>>;
 
 export class TokenFetcher {
   private readonly connection: Connection;
-  private readonly _cache: Record<string, Token> = {};
+  private readonly _cache: Record<string, Token>;
   private readonly providers: MetadataProvider[] = [];
+  private readonly timeoutMs: number;
 
-  private constructor(connection: Connection, cache: Record<string, Token> = {}) {
+  private constructor(connection: Connection, opts: Opts) {
```

Comment:
Is there any reason not to set default values for opts?🤔

Reply:
Nope - added back!